src/prefect/events/clients.py

```diff
@@ -0,0 +1,57 @@
+import abc
+from types import TracebackType
+from typing import List, Optional, Type
+
+from .schemas import Event
+
+
+class EventsClient(abc.ABC):
+    """The abstract interface for a Prefect Events client"""
+
+    @abc.abstractmethod
+    async def emit(self, event: Event) -> None:
+        ...
+
+    async def __aenter__(self) -> "EventsClient":
+        return self
+
+    async def __aexit__(
+        self,
+        exc_type: Optional[Type[Exception]],
+        exc_val: Optional[Exception],
+        exc_tb: Optional[TracebackType],
+    ) -> "EventsClient":
+        return self
+
+
+class NullEventsClient(EventsClient):
+    """An implementation of the Prefect Events client that does nothing"""
+
+    async def emit(self, event: Event) -> None:
+        pass
+
+
+class AssertingEventsClient(EventsClient):
```

Comment:
Should this and the `NullEventsClient` be in `prefect.testing.events`? It may be nice to keep them out of the "maintained" interface.

Reply:
It does make sense to move the `AssertingEventsClient` there, but the `NullEventsClient` will be the one that runs for most people unless they configure another one.  I think that's a "real" one, so to speak.